runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -3179,16 +3189,26 @@ remoteCompile(
          JITServer::ServerMemoryState nextMemoryState = std::get<9>(recv);
          updateCompThreadActivationPolicy(compInfoPT, nextMemoryState);
          }
-      else if (JITServer::MessageType::compilationThreadCrashed == response)
+      else if (JITServer::MessageType::jitDumpPrintIL == response)
          {
+         // A server compilation thread has crashed, and the crashing thread
+         // is generating a JitDump. Use the existing compilation to print IL
+         // of the method causing the crash and then schedule a JitDump recompilation
+         TR::CompilationInfoPerThreadBase::UninterruptibleOperation uop(*compInfoPT);
+         releaseVMAccess(vmThread);
+         while(!handleServerMessage(client, compiler->fej9vm(), response));
+         acquireVMAccessNoSuspend(vmThread);
+         
+         TR_ASSERT_FATAL(
+            response == JITServer::MessageType::compilationThreadCrashed,
+            "MessageType::compilationThreadCrashed is the only valid response after printing IL of the crashing method");
```

Comment:
Is it at all possible to receive a `MessageType::compilationFailure` message from the server while doing the diagnostic recompilation?

Reply:
This message should be received before the recompilation starts., but after IL tracing. I guess it's technically possible to receive `compilationFailure`, if a memory allocation fails or something like that.